### PR TITLE
Attach the instance number to 'cpl' in cpl history file names,

### LINKED ISSUE #2754: Correct the multi-driver coupler history file names
I'm running a multi-driver CAM6 and asking for coupler history files.
The files are being generated with the instance number in the wrong
place in the file names:
CAM6_resource80.cpl.ha2x3h_0001.2010-07-01.nc
instead of 
CAM6_resource80.cpl_0001.ha2x3h.2010-07-01.nc

I've found where this is happening
(seq_hist_writeaux in cime_comp_mod.F90 and seq_hist_mod.F90)
so I can fix this and issue a PR after someone tells me which
CIME version I should clone.  Last time, when I tried to update
my (kdraeder) github cime master from the ESMCI master, 
I found out that it was the wrong one.  
I was supposed to update from something like maint-#.#.
Please point me to the right version.
Thanks,
Kevin

### PR DESCRIPTION
**Pull request by kdraeder**
instead of attaching to the file type (.ha2x, etc.).
Also incidentally includes a fix in case.py
for a problem with fetching GRIDS_SPEC_FILE for
user defined grids (#1798).

Test suite: scripts_regression_tests.py: passes all
Test baseline: 
Test namelist changes: none
Test status: bit for bit

Fixes #2754

User interface changes?: None

Update gh-pages html (Y/N)?:  No

Code review:

### CODE REVIEW

**Comment by jedwards4b** [comment]
@kdraeder you can edit the pr and change the location to be merged into to the maint-5.6 branch.  I can walk you through this monday morning if you like.

**Comment by jedwards4b** [comment]
```
=====================================================================
FAIL: test_pylint_scripts_lib_CIME_case_case_py (__main__.B_CheckCode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 2945, in test
    self.assertTrue(result == "", msg=result)
AssertionError: False is not true : ************* Module CIME.case.case
W:757,31: Unused argument 'user_grid' (unused-argument)
```

File: scripts/lib/CIME/case/case.py [hunk 1]
```diff
@@ -773,11 +773,10 @@ def configure(self, compset_name, grid_name, machine_name=None,
             compset_name, files)
 
         self._components = self.get_compset_components()
+
         #--------------------------------------------
         # grid
         #--------------------------------------------
-        if user_grid is True and gridfile is not None:
```
Comment (billsacks) [inline scripts/lib/CIME/case/case.py]: Was this deletion intentional? It looks unrelated to the other changes in this PR. If it was intentional, can you please describe why you deleted it?

Comment (kdraeder) [reply scripts/lib/CIME/case/case.py]: There should be 2 lines deleted:
-        if user_grid is True and gridfile is not None:
-            self.set_value("GRIDS_SPEC_FILE", gridfile)
because they generated an error when I was implementing
a high resolution SST grid (#1798) and @jedwards4b 
suggested removing those lines.  I need the hi-res in my 
test case, but I suppose for the purpose of this PR I should
use a different test case and keep those 2 lines.
Should I add them back into my branch and push it again,
or go back to an earlier commit and start over with a new,
cleaner PR?

Comment (billsacks) [reply scripts/lib/CIME/case/case.py]: Okay, thanks for the clarification. In an ideal world, this would be in a separate PR, but I'm not going to be a stickler there. If you want to keep it as part of this PR, it's fine with me if you just revert that change on this branch. However, please edit the initial PR comment to document this change (ideally it would also be associated with an issue, but again, I won't be a stickler).

Comment (kdraeder) [reply scripts/lib/CIME/case/case.py]: I edited the initial comment to include a brief description 
of the deletion from case.py.  Thanks for your flexibility.

**Comment by kdraeder** [comment]
@jedwards4b 
I changed the variable names locally, but have not pushed the changes yet
because of the failed check.
I followed the Details link of continuous-integration/travis-ci/pr
and ended up looking for 
/home/travis/cesm/scratch/scripts_regression_test.20180826_144623
to find out why that check failed.  It no longer exists, 
or maybe never did on cheyenne.
Do you have any suggestions for learning about why this check failed?

I also see that Merging is blocked.  I'm guessing that I need to update
the branch of this pull request.  Tell me if there's more to it.

Thanks,
Kevin

**Comment by jedwards4b** [comment]
@kdraeder Please push the changes you want to make.  I'll clean up the travis-ci before I merge.

**Comment by jedwards4b** [comment]
Here is the failing test:
```

======================================================================
FAIL: test_pylint_scripts_lib_CIME_case_case_py (__main__.B_CheckCode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 2945, in test
    self.assertTrue(result == "", msg=result)
AssertionError: False is not true : ************* Module CIME.case.case
W:757,31: Unused argument 'user_grid' (unused-argument)
-----------------------------------
Your code has been rated at 9.99/10
No config file found, using default configuration
```
You can repeat this test locally by running scripts/Tools/code_checker case.py

File: src/drivers/mct/main/cime_comp_mod.F90 [hunk 2]
```diff
@@ -3089,10 +3088,11 @@ subroutine cime_run()
...
                 call t_drvstartf ('driver_rofpost_histaux', barrier=mpicom_CPLID)
                 do eri = 1,num_inst_rof
-                   suffix =  component_get_suffix(rof(eri))
+                   inst_suffix =  component_get_suffix(rof(eri))
                    call seq_hist_writeaux(infodata, EClock_d, rof(eri), flow='c2x', &
-                        aname='r2x'//trim(suffix), dname='domrb', &
-                        nx=rof_nx, ny=rof_ny, nt=1, write_now=t24hr_alarm)
+                        aname='r2x',dname='domrb',inst_suffix=trim(inst_suffix),  &
+                        nx=rof_nx, ny=rof_ny, nt=1, write_now=t6hr_alarm)
+! KDR orig                        nx=rof_nx, ny=rof_ny, nt=1, write_now=t24hr_alarm)
```
Comment (jedwards4b) [inline src/drivers/mct/main/cime_comp_mod.F90]: please remove the comment

File: src/drivers/mct/main/seq_hist_mod.F90 [hunk 3]
```diff
@@ -1041,7 +1042,8 @@ subroutine seq_hist_writeaux(infodata, EClock_d, comp, flow, aname, dname, &
     logical                  :: lwrite_now
     logical                  :: whead, wdata  ! for writing restart/history cdf files
     real(r8)                 :: tbnds(2)
-    character(len=12) :: date_str
+! KDR changed from 12(why 12?) to 16 (YYYY-MM-DD-SSSSS)
```
Comment (jedwards4b) [inline src/drivers/mct/main/seq_hist_mod.F90]: please remove the comment

File: src/drivers/mct/main/seq_hist_mod.F90 [hunk 4]
```diff
@@ -1163,9 +1165,10 @@ subroutine seq_hist_writeaux(infodata, EClock_d, comp, flow, aname, dname, &
              if (present(yr_offset)) then
                 yy = yy + yr_offset
              end if
-             call shr_cal_ymdtod2string(date_str, yy, mm, dd)
-             write(hist_file(found), "(6a)") &
-                  trim(case_name),'.cpl.h',trim(aname),'.',trim(date_str), '.nc'
+! KDR orig             call shr_cal_ymdtod2string(date_str, yy, mm, dd)
```
Comment (jedwards4b) [inline src/drivers/mct/main/seq_hist_mod.F90]: please remove comment

**Comment by kdraeder** [comment]
My PR from several weeks ago failed a python test.
I removed an unused argument (user_grid) from a couple
of routine's calls, and it passed the python test.
I pushed the new code to kdraeder/cime/tree/kdr_fix_cpl_hist_names
on Sep 27.   My understanding was that the push would notify
the people involved with the PR (2782) that there was an update.
Do I need to do more to move it through the pipeline?

Thanks,
Kevin

**Comment by jedwards4b** [comment]
@kdraeder I reviewed that code and made several addtional comments/requests for changes - you have not responded to that request - do you look at the web interface 
https://github.com/ESMCI/cime/pull/2782 ?

**Comment by gold2718** [comment]
Are you sure you pushed successfully? Your branch shows no updates for Sep. 27 (Latest commit 016b83f 8 days ago).

**Comment by jedwards4b** [comment]
@kdraeder thanks.
